src/spikeinterface/core/generate.py

```diff
@@ -550,150 +586,59 @@ def get_traces(
+        # slice channels
+        traces = traces if channel_indices is None else traces[:, channel_indices]
 
         return traces
 
-    def _random_peaks_generator(self, start_frame: int, end_frame: int) -> np.ndarray:
-        """
-        Generate a deterministic trace with sharp peaks for a given range of frames
-        while minimizing memory allocations.
-
-        This function creates a numpy array of deterministic traces between the specified
-        start_frame and end_frame indices.
-
-        The traces exhibit a variety of amplitudes and phases.
-
-        The resulting traces numpy array has a shape (num_samples, num_channels), where num_samples is the
-        number of samples between the start and end frames,
-        and num_channels is the number of channels in the given.
-
-        See issue https://github.com/SpikeInterface/spikeinterface/issues/1413 for
-        a more detailed graphical description.
-
-        Parameters
-        ----------
-        start_frame : int
-            The starting frame index for generating the deterministic traces.
-        end_frame : int
-            The ending frame index for generating the deterministic traces.
 
-        Returns
-        -------
-        np.ndarray
-            A numpy array containing the deterministic traces with shape (num_samples, num_channels).
-
-        Notes
-        -----
-        - This is a helper method and should not be called directly from outside the class.
-        - The 'out' arguments in the numpy functions are important for minimizing memory allocations
-        """
-
-        # Allocate memory for the traces and reuse this reference throughout the function to minimize memory allocations
-        num_samples = end_frame - start_frame
-        traces = np.ones((num_samples, self.num_channels), dtype=self.dtype)
-
-        times_linear = np.arange(start=start_frame, stop=end_frame, dtype=self.dtype).reshape(num_samples, 1)
-        # Broadcast the times to all channels
-        times = np.multiply(times_linear, traces, dtype=self.dtype, out=traces)
-        # Time in the frequency domain; note that frequencies are different for each channel
-        times = np.multiply(
-            times, (2 * np.pi * self.frequencies) / self.sampling_frequency, out=times, dtype=self.dtype
-        )
-
-        # Each channel has its own phase
-        times = np.add(times, self.channel_phases, dtype=self.dtype, out=traces)
-
-        # Create and sharpen the peaks
-        traces = np.sin(times, dtype=self.dtype, out=traces)
-        traces = np.power(traces, 10, dtype=self.dtype, out=traces)
-        # Add amplitude diversity to the traces
-        traces = np.multiply(self.amplitudes, traces, dtype=self.dtype, out=traces)
-
-        return traces
+noise_generator_recording = define_function_from_class(
+    source_class=NoiseGeneratorRecording, name="noise_generator_recording"
+)
 
 
-def generate_lazy_recording(
+def generate_recording_by_size(
```

Comment:
I wish we went all the way to name this "generate_recording_by_memory_size" to be completly specific.